conda_lockfiles/dumpers/conda_lock_v1.py

```diff
@@ -42,31 +61,43 @@ def _record_to_conda_lock_v1_package(
     }
 
 
-def export_to_conda_lock_v1(prefix: str, lockfile_path: str | None) -> None:
-    prefix_data = PrefixData(prefix)
-    packages = [
-        _record_to_conda_lock_v1_package(p, context.subdir)
-        for p in prefix_data.iter_records()
-    ]
-    channel_urls = {(p.schannel) for p in prefix_data.iter_records()}
-    metadata = {
-        "content_hash": {},
-        "channels": [{"url": url, "used_env_vars": []} for url in channel_urls],
-        "platforms": [context.subdir],
-        "sources": [""],
-        "time_metadata": {
-            "created_at": datetime.datetime.now(datetime.timezone.utc).strftime(
-                "%Y-%m-%dT%H:%M:%SZ"
-            ),
-        },
-        "custom_metadata": {
-            "created_by": "conda-lockfiles",
-        },
-    }
-    output = {
+def _to_dict(env: Environment) -> dict[str, Any]:
+    missing_urls = [package for package in env.explicit_packages if package.url is None]
+    if missing_urls:
+        raise EnvironmentExportNotSupported(
+            FORMAT,
+            f"The following packages have no URL: {dashlist(missing_urls)}",
+        )
+    timestamp = datetime.now(timezone.utc).strftime(TIMESTAMP)
+    return {
         "version": 1,
-        "metadata": metadata,
-        "package": sorted(packages, key=lambda x: x["name"]),
+        "metadata": {
+            "content_hash": {},
+            "channels": [
+                {
+                    "url": channel,
+                    "used_env_vars": [],
+                }
+                for channel in env.config.channels
+            ],
+            "platforms": [env.platform],
+            "sources": [""],
+            "time_metadata": {"created_at": timestamp},
+            "custom_metadata": {"created_by": "conda-lockfiles"},
```

Comment:
Should we include the `conda-lockfiles` version?

```suggestion
            "custom_metadata": {"created_by": f"conda-lockfiles {__version__}"},
```